contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol

```diff
@@ -102,23 +105,37 @@ contract TermsOfServiceAllowList is Routable, ITermsOfServiceAllowList {
     }
 
     // Add recipient to the allow list
-    s_allowedSenders[recipient] = true;
+    s_allowedSenders.add(recipient);
   }
 
   /**
    * @inheritdoc ITermsOfServiceAllowList
    */
-  function isAllowedSender(address sender) public view override returns (bool) {
+  function getAllAllowedSenders() external view override returns (address[] memory) {
+    return s_allowedSenders.values();
+  }
+
+  /**
+   * @inheritdoc ITermsOfServiceAllowList
+   */
+  function hasAccess(address sender) external view override returns (bool) {
+    if (!s_config.enabled) {
+      return true;
+    }
+    return s_allowedSenders.contains(sender);
+  }
+
+  function hasAccess(address user, bytes calldata /* data */) external view returns (bool) {
```

Comment:
why two methods?

Reply:
`hasAccess` has that data field, which is fairly annoying to call.
Easier to allow the overload.

Reply:
I'm with Bolek, I think just one method where you pass empty bytes is better. I'm pretty sure that adding another public method will take up more bytecode space.

Reply:
Changed

Reply:
How much more gas does `haveAccess` require if we need to pass this unused data argument?  Probably not much, but it the interest of gas golf it might be a good optimization.